hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -397,6 +401,9 @@ protected HoodieMergeHandle getUpdateHandle(String partitionPath, String fileId,
       mergeHandle.setPartitionFields(partitionFields);
       mergeHandle.setPartitionValues(partitionValues);
     }
+    if (readerContextFactory != null && mergeHandle instanceof FileGroupReaderBasedMergeHandle) {
```

Comment:
anyways, constructor for FileGroupReaderBasedMergeHandle is diff from HoodieWriteMergeHandle. 
so, do you think we can add a new arg to constructor as Option<SerializableFunc> which can assist with setting the reader context from within FileGroupReaderBasedMergeHandle class only, rather than special casing here at L405.

Reply:
not to strong on the suggestion though. Will let you decide.

Reply:
I did this before but @danny0405 reverted it.